### ISSUE TITLE
listing sub projects

### ISSUE THREAD

**Issue by V3ckt0r** [cvbarros/go-teamcity#66]
Looking at the Project struct I can't see a way of listing sub project to a project. Is there another way this can be done that I am missing? I see there is a `ParentProject` but no  child/sub project.

**Comment by V3ckt0r** [comment]
To provide a little more info. I'm looking for the following api
GET /app/rest/projects

**Comment by cvbarros** [comment]
That's right @V3ckt0r , there is no support yet for ad-hoc project querying.
That would require changes to the `ProjectService` to accept `locators`.

I played a bit in local trying to come up with a generic locators implementation for several endpoints but I wasn't happy about the design so I dropped it a while ago, focusing more on other features.

However, it won't be hard to implement some locators support for projects and a `GetByLocator` in `projects/` endpoint.

Would you please elaborate a bit more if is there any locators/filters you'd like implemented?

**Comment by JaSei** [comment]
Hi, I need to list all projects too. So my current solution is to use https://github.com/kapitanov/go-teamcity where is a list of projects implemented (`GetProjects`).

**Comment by JaSei** [comment]
This feature will be great. For me, it will be nice to be able to list all VcsRoots.
